### ISSUE TITLE
Add region landmarks to Products Details Page

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13518]
**Describe the bug**

- Screen Reader users must be able to navigate by landmarks, currently, there aren't many nor labelled, therefore relevant sections on the page should be identified with meaningful landmarks names.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS).
2. Bring up the Rotor or Element List Menu.
3. Choose Landmarks view.
4. See absence of relevant landmarks.

**Expected behavior**

- Use only predefined landmarks https://wiki.wdf.sap.corp/wiki/display/spar/Landmarks
- `<cx-carousel role="region" aria-label="Carousel, Images of product name">`
- `<cx-tab-paragraph-container role="region" aria-label="Tab group with more product details">`

**Comment by BenjaminEmiliani** [comment]
On hold until after #12151

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.
<img width="1182" alt="Screen Shot 2022-03-14 at 2 16 55 PM" src="https://user-images.githubusercontent.com/45409515/158235863-ca30c783-21ba-493c-b308-405effd2ce94.png">
